src/number/streaming.rs

```diff
@@ -1538,6 +1538,79 @@ where
   }
 }
 
+#[inline]
+#[doc(hidden)]
+pub fn varint<I, E: ParseError<I>, R>(i: I) -> IResult<I, R, E>
+where
+  I: Slice<RangeFrom<usize>> + InputIter<Item = u8> + InputLength + Copy,
+  R: std::ops::AddAssign<R> + std::ops::Shl<usize, Output = R> + Default + From<u8>,
+{
+  let mut result = R::default();
+  let mut ofs: usize = 0;
+  let mut remainder = i;
+  loop {
+    let (i, byte) = le_u8(remainder)?;
+    result += (<R as From<u8>>::from(byte & 0x7F)) << ofs;
+    if byte < 0x80 {
```

Comment:
I believe this should exit if enough bytes for the target data type have been collected. Otherwise the right shift or addition will overflow.